code/core/src/preview-api/modules/preview-web/render/StoryRender.test.ts

```diff
@@ -153,17 +152,14 @@ describe('StoryRender', () => {
     );
 
     await render.renderToElement({} as any);
-    expect(actualMount).toHaveBeenCalled();
+    expect(mountSpy).toHaveBeenCalledOnce();
   });
 
-  it('does not call mount if play function destructures mount', async () => {
-    const actualMount = vi.fn(async (context) => {
-      await context.renderToCanvas();
-      return context.canvas;
-    });
+  it('does not call mount twice if mount called in play function', async () => {
```

Comment:
@tmeasday I changed the assertion. As we prevent calling mount twice, if the user wants to mount in play.

It will always be called once. If not, we throw an error.

Reply:
We kind of `auto-mount` before the play function, except when you destructure mount. 
So this test ensures that `mount` is only called once in that scenario and not twice.

Reply:
I see. What if I have a play function like this thought?

```js
play: async ({ mount }) => {
  // some setup
  await mount();
  // who knows what 
  await mount();
  // etc.
}
```

Reply:
I agree, let's add a test that we throw in that scenario! But I agree this change is an improvement.